src/main/java/hanieum/conik/domain/member/dto/MemberLoginResponse.java

```diff
@@ -2,10 +2,15 @@
 
 import io.swagger.v3.oas.annotations.media.Schema;
 
+import java.util.Optional;
+
 public record MemberLoginResponse(
         @Schema(description = "토큰 정보")
         TokenInfo tokenInfo,
 
         @Schema(description = "멤버 정보")
-        MemberInfo memberInfo
-) { }
+        MemberInfo memberInfo,
+
+        @Schema(description = "멤버가 속한 기업 id", nullable = true)
+        Optional<Long> companyId
```

Comment:
from 이나 of 등의 정적 팩토리 메서드 를 통해서 생성해주는 로직을 만들고 이걸 AuthService에서 사용할 수도 있을 것 같아요

Reply:
```java
public record MemberLoginResponse(
    TokenInfo      tokenInfo,
    MemberInfo     memberInfo,
    Optional<Long> companyId
) {
    public static MemberLoginResponse of(Member member, TokenInfo tokenInfo) {
        Optional<Long> companyId = member.getRole() == MemberRole.INDIVIDUAL
            ? Optional.empty()
            : Optional.of(member.getCompanyId());
        return new MemberLoginResponse(tokenInfo, MemberInfo.from(member), companyId);
    }
}
```
로그인 로직에서 이렇게 메서드를 만들어 사용할까 고민하긴 했었는데 이렇게 되면 ROLE에 따른 companyId의 반환값이 정해지는 비즈니스 로직이 record에 들어가게 된다고 생각해서 AuthService에 로직을 빼두었습니다. 
dto는 단순 데이터와 매핑만 책임지게 하려고 했는데 뭐가 더 좋은 방법인지는 잘 모루겠숴요..

Reply:
그러면 파라미터로 Member member, TokenInfo tokenInfo를 갖는 of와
파라미터로 Member member, TokenInfo tokenInfo, Long CompanyId를 갖는 of 이렇게 두개를 선언해두고
불러만 쓰는 건 어떤가요?

Reply:
@SeongHo5356 일반 멤버인 경우랑 기업 멤버인 경우를 확실하게 구분해서 표현하고 싶어서 그런 걸까요??

Reply:
```java
public static MemberLoginResponse of(TokenInfo tokenInfo, MemberInfo memberInfo, Optional<Long> companyId) {
                return new MemberLoginResponse(tokenInfo, memberInfo, companyId.isEmpty() ? Optional.empty() : companyId);
        }
```
따로 두지 않고 이렇게 구현하는건 어떤가요??

Reply:
Optional은 “주로 메서드의 반환 타입으로 사용하도록 의도됐다”고 JavaDoc에 명시돼 있다고 하네요. 따라서, 필드나  파라미터 타입으로 쓰라기 보단 결과가 없을 수 있는 상황에서 null 대신 쓰는 용도로 알맞을 것 같아요. 아래처럼 구현하는 게 좋을 것 같네요!

```suggestion
{
    // 1개로 처리
    public static MemberLoginResponse of(TokenInfo token, Member member) {
        Long cid = (member.getRole() == MemberRole.INDIVIDUAL) ? null : member.getCompanyId();
        return new MemberLoginResponse(token, MemberInfo.from(member), cid);
    }

    // 역할 구분
    public static MemberLoginResponse individual(TokenInfo token, Member member) {
        return new MemberLoginResponse(token, MemberInfo.from(member), null);
    }
    public static MemberLoginResponse corporate(TokenInfo token, Member member, Long companyId) {
        return new MemberLoginResponse(token, MemberInfo.from(member), Objects.requireNonNull(companyId));
    }
}
```